controllers/Sitemap.js

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable no-restricted-syntax */
+/* ------------------------------------------------------------------------------------------------
+ * Copyright (c) mgm security partners GmbH. All rights reserved.
+ * Licensed under the AGPLv3 License. See LICENSE.md in the project root for license information.
+ * ------------------------------------------------------------------------------------------------*/
+
+const xmlbuilder = require('xmlbuilder');
+const TagService = require('../service/TagService');
+const TestcaseService = require('../service/TestcaseService');
+
+module.exports.sitemapIndex = async (req, res) => {
+  try {
+    const PathDefault = 'https://canitrust.in/';
+    const TagDefault = 'https://canitrust.in/tag/';
+    const obj = {
+      urlset: {
+        '@xmlns': 'http://www.sitemaps.org/schemas/sitemap/0.9',
+        url: [],
+      },
+    };
+    const PathData = {
+      testcases: await TestcaseService.getAllPageTestcase(),
+    };
+    const homepageElement = {
+      loc: PathDefault,
+      lastmod: formatDate(
+        PathData.testcases[PathData.testcases.length - 1].date_created
+      ),
+      changefreq: 'Weekly',
+    };
+    obj.urlset.url.push(homepageElement);
+    for (const item of PathData.testcases) {
+      // pathList.push(item.path,item.date_created);
+      const elementPath = {
+        loc: PathDefault.concat(item.path),
+        lastmod: formatDate(item.date_created),
+        changefreq: 'Monthly',
+      };
+      obj.urlset.url.push(elementPath);
+    }
+    const TagData = {
+      tags: await TagService.getAllTag(),
+    };
+    for (const item of TagData.tags) {
+      const elementTag = {
+        loc: TagDefault.concat(item.tagText),
+        lastmod: formatDate('Mon July 1,2019'),
```

Comment:
Improvement: can we set `lastmod` of a tag with the date of lastest testcase of the tag?